File: ios/Classes/lib/awesome_notifications_core/exceptions/AwesomeNotificationsException.swift
public class AwesomeNotificationsException: Error {
    public let className:String
    public let code:String
    public let message:String
    public let detailedCode:String
    
    init(className:String, code:String, message:String, detailedCode:String){
        self.className = className
        self.code = code
        self.message = message
        self.detailedCode = detailedCode
    }
}

File: android/src/main/java/me/carda/awesome_notifications/awesome_notifications_core/exceptions/AwesomeNotificationsException.java
package me.carda.awesome_notifications.awesome_notifications_core.exceptions;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

public class AwesomeNotificationsException extends Exception {

    final String code;

    AwesomeNotificationsException(
            @NonNull String code,
            @NonNull String message
    ){
        super(message);
        this.code = code;
    }

    AwesomeNotificationsException(
            @NonNull String code,
            @NonNull String message,
            @NonNull StackTraceElement[] stackTraceElement
    ){
        super(message);
        this.code = code;
        this.setStackTrace(stackTraceElement);
    }

    AwesomeNotificationsException(
            @NonNull String code,
            @NonNull String message,
            @NonNull Exception originalException
    ){
        super(message);
        this.code = code;
    }

    @NonNull
    public final String getCode(){
        return code;
    }
}